include/llama/mapping/One.hpp

```diff
@@ -35,14 +44,39 @@ namespace llama::mapping
 
         LLAMA_FN_HOST_ACC_INLINE constexpr auto blobSize(std::size_t) const -> std::size_t
         {
-            return sizeOf<RecordDim>;
+            return flatSizeOf<typename Flattener::FlatRecordDim, AlignAndPad, false>; // no tail padding
         }
 
         template <std::size_t... RecordCoords>
         LLAMA_FN_HOST_ACC_INLINE constexpr auto blobNrAndOffset(ArrayDims) const -> NrAndOffset
         {
-            constexpr auto offset = offsetOf<RecordDim, RecordCoord<RecordCoords...>>;
+            constexpr std::size_t flatIndex =
+#ifdef __NVCC__
+                Flattener{}.template flatIndex<RecordCoords...>;
+#else
+                Flattener::template flatIndex<RecordCoords...>;
+#endif
+            constexpr auto offset = flatOffsetOf<typename Flattener::FlatRecordDim, flatIndex, AlignAndPad>;
             return {0, offset};
         }
+
+    private:
+        using Flattener = FlattenRecordDim<T_RecordDim>;
     };
+
+    /// One mapping preserving the alignment of the field types by inserting padding.
+    /// \see One
+    template <typename ArrayDims, typename RecordDim>
+    using AlignedOne = One<ArrayDims, RecordDim, true, FlattenRecordDimInOrder>;
+
+    /// One mapping preserving the alignment of the field types by inserting padding and permuting the field order to
+    /// minimize this padding.
+    /// \see One
+    template <typename ArrayDims, typename RecordDim>
+    using MinAlignedOne = One<ArrayDims, RecordDim, true, FlattenRecordDimMinimizePadding>;
```

Comment:
Is `One<ArrayDims, RecordDim, false, FlattenRecordDimMinimizePadding>` missing on purpose?

Reply:
That is a valid mapping, but I do not see the use case. `One<..., true, ...>` adds padding to respect alignment, so adding field permutations make sense, to play with this padding.
`One<..., false, ...>` does not add padding, so permuting the fields has a far smaller effect. So I want to advertise it less.